tools/default.nix

```diff
@@ -0,0 +1,65 @@
+{ pkgs ? import <nixpkgs> {} , withEditor ? false }:
+
+pkgs.stdenv.mkDerivation rec {
+  name = "openclonk";
+
+  gitRef = pkgs.lib.commitIdFromGitRepo ../.git;
+
+  src = builtins.filterSource (path: type: ! builtins.elem (baseNameOf path) [
+    ".git" # leave out .git as it changes often in ways that do not affect the build
+    "default.nix" # default.nix might change, but the only thing that matters is what it evaluates to, and nix takes care of that
+    "result" # build result is irrelevant
+    "build"
+  ]) ./..;
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = "format";
+
+  nativeBuildInputs = with pkgs; [ cmake ];
+
+  dontStrip = true;
+
+  buildInputs = with pkgs; [
+    SDL2 libvorbis libogg libjpeg libpng freetype glew tinyxml
+    openal freealut pkgconfig
+  ] ++ stdenv.lib.optional withEditor qt5.full;
+
+  cmakeFlags = [ "-DCMAKE_AR=${pkgs.gcc-unwrapped}/bin/gcc-ar" "-DCMAKE_RANLIB=${pkgs.gcc-unwrapped}/bin/gcc-ranlib" ];
+
+  preConfigure = ''
+    sed s/REVGOESHERE/''${gitRef:0:12}/ > cmake/GitGetChangesetID.cmake <<EOF
```

Comment:
Ah, I think I didn't experience that because git wasn't available in my build so it didn't go into the GIT_FOUND branch.

Do you think we should set `C4REVISION_DIRTY` and `C4REVISION_TS` too?